databroker/core.py

```diff
@@ -249,9 +261,17 @@ def get_table(mds, fs, headers, fields=None, fill=True, convert_times=True,
         # cache these attribute look-ups for performance
         start = header['start']
         stop = header['stop']
+        descriptors = header['descriptors']
         if stop is None:
             stop = {}
-        for descriptor in header['descriptors']:
+
+        # shim for back-compat with old data that has no 'primary' descriptor
+        if not [d for d in descriptors if d.get('name') == 'primary']:
```

Comment:
Maybe in the future, but I don't see this as any kind of bottleneck yet. The number of descriptors per run is not large and will probably never be more than 10s, so we just fetch them all and filter in Python.